drivers/soundwire/cadence_master.c

```diff
@@ -1703,10 +1704,7 @@ int cdns_set_sdw_stream(struct snd_soc_dai *dai,
 
 	if (stream) {
 		/* first paranoia check */
-		if (direction == SNDRV_PCM_STREAM_PLAYBACK)
-			dma = dai->playback_dma_data;
-		else
-			dma = dai->capture_dma_data;
+		dma = snd_soc_dai_get_dma_data(dai, stream);
```

Comment:
does this work?

The 'stream' passed to cdns_set_sdw_stream() is of type 	"struct sdw_stream_runtime *stream;"

the snd_soc_dai_get_dma_data uses stream as const struct snd_pcm_substream *ss

not sure how this might work?

There was a good reason why we did this manual check, it was intentional IIRC